hooks/addon

```diff
@@ -108,10 +114,53 @@ reissue() {
   fi
 
   secret=$vault/openvpn/certs/users/$email
-  safe x509 reissue $secret
+  safe x509 reissue --signed-by $vault/openvpn/certs/ca $secret
   safe x509 show $secret
 }
 
+generate_vpn_config() {
+  email=${1:-}
+  if [[ -z ${email} ]]; then
+    echo >&2 "USAGE: genesis do <env> generate-vpn-config user@email.addr.ess"
+    exit 1
+  fi
+  if ! safe --quiet check $vault/openvpn/certs/users/$email; then
+    echo "Generating new openvpn client certificate for $email"
+    issue $email
+  fi
+  echo "Generating $email.ovpn config file..."
+  if [[ $ips == null ]]; then
+    echo "Failed to get VPN External IP from BOSH or Params - check your connection to BOSH or params file"
+    exit 1
+  fi
+  (
+    echo "client"
+    echo "dev tun"
+    echo "proto tcp"
+    echo "remote $(lookup params.vpn_external_ip  $ips) $(lookup params.vpn_external_port  443)"
+    echo "resolv-retry infinite"
+    echo "nobind"
+    echo "persist-key"
+    echo "persist-tun"
+    echo "mute-replay-warnings"
+    echo "remote-cert-tls server"
+    echo "verb 3"
+    echo "mute 20"
+    echo "tls-client"
+    echo "cipher AES-256-CBC"
+    echo "keysize 256"
+    echo "<ca>"
+    safe get $vault/openvpn/certs/ca:certificate
+    echo "</ca>"
+    echo "<cert>"
+    safe get $vault/openvpn/certs/users/$email:certificate
+    echo "</cert>"
+    echo "<key>"
+    safe get $vault/openvpn/certs/users/$email:key
+    echo "</key>"
+  ) >$GENESIS_ROOT/$email.ovpn
```

Comment:
Does this need to go to a file, or does can it just go to stdout?

Reply:
I have two concerns:

1. The files always get created in the GENESIS_ROOT, which may not always be the current working directory.  `genesis -C some/dir` for example, lets you run commands _as if_ you were chdir'd to `some/dir`, so the ovpn files would be created there.  This seems surprising to me,
2. There are credentials in these files, and since they exist in-working-copy, there is a pretty high chance that keys will get committed to GitHub.  At least with file redirection, _the operator_ is primarily responsible for creating (and then commiting) that file.

If we keep the file-creation, we should at least allow the following scenarios:

- Multiple email addresses on a single invocation, i.e.: `genesis do generate-vpn-config {dave,james,other-dave}@example.com`
- Allow all known users to be re-generated / re-inflated, without naming names, as in `genesis do generate-vpn-config all` or just `genesis do generate-vpn-config`

Reply:
Makes sense -> updated to throw configs to stdout and info to stderr. The config files are to be more or less ephemeral once loaded into a vpn client of choice.